### ISSUE TITLE
No build or runtime errors if missing Microsoft.Azure.Functions.Worker package reference

### ISSUE THREAD

**Issue by DannyMeister** [Azure/azure-functions-dotnet-worker#1386]
I migrated an in-proc Azure Function project to isolated, and somehow missed adding the actual `Microsoft.Azure.Functions.Worker` package, though I added all the other necessary `Microsoft.Azure.Functions.Worker.*` packages (don't ask me how I did that 🤷‍♂️).

This is potentially similar to this reported issue: #1347 

## Expected Behavior

I would have expected a runtime error in the console.  Even better, a static analyzer or a compile time error could have alerted me to the issue.

## Actual Behavior
![image](https://user-images.githubusercontent.com/7937829/221266423-e376e6f1-6342-4256-a9e3-ef52fe6c4ae8.png)

Console log shows that my triggers are found.
Calling my HttpTrigger shows execution of my trigger starting.  Yet code is never executed, breakpoints never hit, and no error is produced in the log.  The request just hangs there.

Even with --verbose flag (not pictured) I didn't get any output that clued me in to what was happening.  Instead, I finally stumbled on it during file comparisons with another working project.

## Environment

VS 2022 version 17.5
Windows 10 21H2 (19044.2486)
Microsoft.Azure.Functions.Worker.Core 1.8.1 (obviously, i can't give the Microsoft.Azure.Functions.Worker version I wasn't using :D 
)
AzureFunctionTools Release 4.32.0
Core Tools Version:       4.0.5030 Commit hash: N/A  (64-bit)
Function Runtime Version: 4.15.2.20177

**Comment by kshyju** [comment]
The `Microsoft.Azure.Functions.Worker` package is a must have package you need to refer for a dotnet isolated function. A proper dotnet isolated function will not even build successfully without this package because the startup code (in the Program.cs) relies methods from this package.
 
> I migrated an in-proc Azure Function project to isolated.

Did you have a main method / Program.cs ? If you did not, I would not agree that your migration was done.  If you were following a guide for the migration, can you share that  please?

**Comment by kshyju** [comment]
Also, curious to know the exact state of your project when you thought migration was done, but app was not working. This will help us to determine what tooling support is appropriate (ex: analyzers/ code fix providers) for the migration case.

**Comment by DannyMeister** [comment]
I did have a Program.cs with a `HostBuilder` that I configured and called `host.Build()` and `await host.RunAsync()`.  I would agree though that I wasn't fully done with migration due to the fact I was missing a critical package, yet there were no good clues for me.  I did not need to include `Microsoft.Azure.Functions.Worker` in order to compile.  I presume the types I needed for compile were in the `Microsoft.Azure.Functions.Worker.Core`, `Microsoft.Azure.Functions.Worker.Http`, etc. packages, or maybe there was something I hadn't noticed hadn't been migrated yet that was still referencing something from the old Azure functions packages, but I have moved on quite a bit from there and will have to scrounge up a bit of time to go back and recreate this scenario from a simple shareable repository.

As far as a guide, I tried to use mainly the official docs to get an understanding of the parts necessary to get this working, but I did also find the following page helpful: https://blog.upperdine.dev/migrating-your-azure-functions-to-run-in-an-isolated-process

Neither place would have led me to exclude the package.  I presume that was an accident on my part at some point... maybe when I was removing old in-proc related packages I accidentally deleted a line too many.
